alembic/versions/735063d71b57_added_document_uploads_and_team_deletion.py

```diff
@@ -0,0 +1,56 @@
+"""Added document uploads and team deletion
+
+Revision ID: 735063d71b57
+Revises: c31b1a94a27d
+Create Date: 2017-11-26 00:16:52.930412
+"""
+
+# revision identifiers, used by Alembic.
+revision = '735063d71b57'
+down_revision = 'c31b1a94a27d'
+branch_labels = None
+depends_on = None
+
+from alembic import op
+import sqlalchemy as sa
+import sideboard.lib.sa
+
+try:
+    is_sqlite = op.get_context().dialect.name == 'sqlite'
+except:
+    is_sqlite = False
+
+if is_sqlite:
+    op.get_context().connection.execute('PRAGMA foreign_keys=ON;')
+    utcnow_server_default = "(datetime('now', 'utc'))"
+else:
+    utcnow_server_default = "timezone('utc', current_timestamp)"
+
+
+def sqlite_column_reflect_listener(inspector, table, column_info):
+    """Adds parenthesis around SQLite datetime defaults for utcnow."""
+    if column_info['default'] == "datetime('now', 'utc')":
+        column_info['default'] = utcnow_server_default
+
+sqlite_reflect_kwargs = {
+    'listeners': [('column_reflect', sqlite_column_reflect_listener)]
+}
+
+
+def upgrade():
+    op.create_table('mits_document',
+    sa.Column('id', sideboard.lib.sa.UUID(), nullable=False),
+    sa.Column('team_id', sideboard.lib.sa.UUID(), nullable=False),
+    sa.Column('filename', sa.Unicode(), server_default='', nullable=False),
+    sa.Column('description', sa.Unicode(), server_default='', nullable=False),
+    sa.ForeignKeyConstraint(['team_id'], ['mits_team.id'], name=op.f('fk_mits_document_team_id_mits_team')),
+    sa.PrimaryKeyConstraint('id', name=op.f('pk_mits_document'))
+    )
+    op.add_column('mits_team', sa.Column('deleted', sa.Boolean(), server_default='False', nullable=False))
+    op.add_column('mits_team', sa.Column('duplicate_of', sideboard.lib.sa.UUID(), nullable=True))
```

Comment:
This won't actually pass our unit tests (even though it looks like it does on Travis, because we skip the migration tests). See [this issue](https://github.com/magfest/ubersystem/issues/2517) and [this comment](https://github.com/magfest/ubersystem/pull/2658#pullrequestreview-50085990) for more discussion.

In short, we have to do a lot of bending over backwards to accommodate SQLite. You can create and drop tables, but you can't directly alter them. You can fix the migration so the unit tests will work like so:
```python
    if is_sqlite:
        with op.batch_alter_table('mits_team', reflect_kwargs=sqlite_reflect_kwargs) as batch_op:
            batch_op.add_column(sa.Column('deleted', sa.Boolean(), server_default='False', nullable=False))
            batch_op.add_column(sa.Column('duplicate_of', sideboard.lib.sa.UUID(), nullable=True))
    else:
        op.add_column('mits_team', sa.Column('deleted', sa.Boolean(), server_default='False', nullable=False))
        op.add_column('mits_team', sa.Column('duplicate_of', sideboard.lib.sa.UUID(), nullable=True))
```

I'm not super happy about the situation, but it has caught a few bugs for me, so I am tolerating it 😅

Reply:
Ahh, good catch, thanks.  I've just fixed this and will make sure to do that in the future anytime I add a migration.